qiskit/visualization/timeline/interface.py

```diff
@@ -0,0 +1,405 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=missing-return-type-doc
+
+"""Qiskit timeline drawer.
+
+This module provides a common user interface of the timeline drawer.
+The `draw` function takes a transpiled circuit program to visualize with a stylesheet
+along with several control arguments.
+The drawer canvas object is internally initialized from the input data and
+the configured canvas is passed to the one of plotter APIs to generate a visualization data.
+"""
+
+from typing import Optional, Dict, Any, List, Tuple
+
+from qiskit import circuit
+from qiskit.visualization.exceptions import VisualizationError
+from qiskit.visualization.timeline import types, core, stylesheet
+
+
+def draw(program: circuit.QuantumCircuit,
+         style: Optional[Dict[str, Any]] = None,
+         time_range: Tuple[int, int] = None,
+         disable_bits: List[types.Bits] = None,
+         show_clbits: Optional[bool] = None,
+         show_idle: Optional[bool] = None,
+         show_barriers: Optional[bool] = None,
+         show_delays: Optional[bool] = None,
+         show_labels: bool = True,
+         plotter: Optional[str] = types.Plotter.MPL.value,
+         axis: Optional[Any] = None,
+         filename: Optional[str] = None):
+    r"""Generate visualization data for scheduled circuit programs.
+
+    Args:
+        program: Program to visualize. This program should be a `QuantumCircuit` which is
+            transpiled with gate time information before visualization.
+        style: Stylesheet options. This can be dictionary or preset stylesheet classes. See
+            :py:class:`~qiskit.visualization.timeline.stylesheets.IqxStandard`,
+            :py:class:`~qiskit.visualization.timeline.stylesheets.IqxSimple`, and
+            :py:class:`~qiskit.visualization.timeline.stylesheets.IqxDebugging` for details of
```

Comment:
I think IQX should be capitalized in each of these, because it's an acronym
https://stackoverflow.com/questions/2853531/how-do-you-pep-8-name-a-class-whose-name-is-an-acronym

Reply:
Thanks for catching this. `IQX` style series was originally temporal name and I'm still using it, i.e. circuit visualizer gate color is based on quantum experience circuit composer. But timeline and pulse visualizer don't exist in IQX, so the stylesheets for timeline don't need to be `IQX`. Any suggestion for new name or IQX is good?

Reply:
I think it's fine (we can introduce timing into IQX later). I would just capitalize it.